src/lib/npm.js

```diff
@@ -22,32 +22,35 @@ async function getNpmToken({npm, options}) {
   };
 
   const uri = url.resolve(npm.registry, '-/user/org.couchdb.user:' + encodeURIComponent(npm.username));
+  let token;
 
-  const {err, token} = await new Promise(resolve => {
-    client.request(uri, {method: 'PUT', body}, async (err, parsed, raw, response) => {
-      if (err && err.code === 'E401' && response.headers['www-authenticate'] === 'OTP') {
-        await askForOTP(uri, body, npm);
-        resolve({token: npm.token});
-      } else if (err && err.code === 'E409') {
-        // Some registries (Sinopia) return 409 for existing users, retry using authenticated call
-        return client
-          .requestAsync(uri, {
-            authed: true,
-            method: 'PUT',
-            auth: {username: npm.username, password: npm.password},
-            body,
-          })
-          .then(res => resolve(res))
-          .catch(err => resolve({err}));
-      } else if (err) {
-        resolve({err});
-      } else {
-        resolve(response.body);
-      }
+  try {
+    [{token}] = await pify(client.request.bind(client), {multiArgs: true})(uri, {
+      method: 'PUT',
+      body,
     });
-  });
+  } catch (err) {
+    const [error, , , response] = err;
+
+    if (error.code === 'E401' && response.headers['www-authenticate'] === 'OTP') {
+      await askForOTP(uri, body, npm);
+      ({token} = npm);
+    } else if (error.code === 'E409') {
+      // Some registries (Sinopia) return 409 for existing users, retry using authenticated call
+      try {
+        ({token} = await pify(client.request.bind(client))(uri, {
+          authed: true,
+          method: 'PUT',
+          auth: {username: npm.username, password: npm.password},
+          body,
+        }));
+      } catch (err) {
+        log.verbose(`Error: ${err}`);
+      }
+    }
+    log.verbose(`Error: ${error}`);
```

Comment:
I would only log this error if we can not handle it. Alternatively this should be moved to the top of the catch clause.

Reply:
I now log this error only if not handled in of the previous cases.